readme/Readme.scalatex

```diff
@@ -322,6 +335,89 @@
                     To explicitly isolate a cut to one branch of a parser, place that branch within @hl.scala{NoCut}.  Cuts within that branch will prevent backtracking inside that branch, but if that branch fails alternate branches will be tried as normal.
 
                 @hl.ref(tests/"ExampleTests.scala", Seq("'composenocut", ""))
+        @sect{Unapply}
+            @p
+                @hl.scala{Parser} class has (as many other classes in Scala) a convenient method @hl.scala{unapply}
+                which allows to do pattern matching using parsers (like regular expressions) in your code.
+
+                @hl.ref(tests/"ExampleTests.scala", "val capture1", "val captureOpt")
+
+
+    @sect{Writing Byte Parsers}
```

Comment:
Let's move this, together with the example byte parsers, to a new top-level section next to `Fastparse` and `ScalaParse` and `Changelog`. For now, the people using byte-parsers will probably be pretty different from the people using string-parsers, so giving them a separate top-level section probably makes sense.